cleanlab/multiannotator.py

```diff
@@ -0,0 +1,436 @@
+    return_annotator_stats: bool = False,
+    verbose: bool = True,
+    kwargs: dict = {},
+) -> Union[pd.DataFrame, Tuple[pd.DataFrame, pd.DataFrame]]:
+    """Returns label quality scores for each example for each annotator.
+    This function is for multiclass classification datasets where examples have been labeled by
+    multiple annotators (not necessarily the same number of annotators per example).
+    It computes quality scores for each annotator's labels and other useful values like how
+    confident we are that the current consensus label is actually correct.
+    The score is between 0 and 1; lower scores
+    indicate labels less likely to be correct. For example:
+    1 - clean label (the given label is likely correct).
+    0 - dirty label (the given label is unlikely correct).
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        labels_multiannotator[n][m] = given label for n-th example by m-th annotator.
+        For a dataset with K classes, the given labels must be an integer in 0, 1, ..., K-1 or
+        np.nan if this annotator did not label a particular example. Column names should correspond to the annotators' ID.
+
+    pred_probs : np.array
+        An array of shape ``(N, K)`` of model-predicted probabilities,
+        ``P(label=k|x)``. Each row of this matrix corresponds
+        to an example `x` and contains the model-predicted probabilities that
+        `x` belongs to each possible class, for each of the K classes. The
+        columns must be ordered such that these probabilities correspond to
+        class 0, 1, ..., K-1.
+        **Caution**: `pred_probs` from your model must be out-of-sample!
+        You should never provide predictions on the same examples used to train the model,
+        as these will be overfit and unsuitable for finding label-errors.
+        To obtain out-of-sample predicted probabilities for every datapoint in your dataset, you can use :ref:`cross-validation <pred_probs_cross_val>`.
+        Alternatively it is ok if your model was trained on a separate dataset and you are only evaluating
+        data that was previously held-out.
+
+    consensus_method : str or List[str]
+        For each example, which method should be used to aggregate labels from multiple annotators into a single consensus label.
+        Options include:
+        - ``"majority"``: means consensus labels are reached via a simple majority vote among annotators, with ties broken via ``pred_probs``.
+        A List may be passed if you want to consider multiple methods for producing consensus labels.
+        If a List is passed, then the 0th element of List is the method used to produce columns "consensus_label", "quality_of_consensus", "annotator_agreement" in the returned DataFrame.
```

Comment:
I don't think method should affect "annotator_agreement" for now?

Reply:
It does because `annotator_agreement` is dependent on the consensus, which is dependent on the consensus method specified. 
There might be a better way to write this, it might currently be confusing as we do not return `annotator_agreement` scores for the other methods (although it could be implicitly used in the calculation of `quality_of_consensus`)

Reply:
I think we resolved this in discussion but feel free to unresolve if not